README.md

```diff
@@ -1,77 +1,185 @@
-- **CheckDependencies**: Looks for dependency files (e.g., `requirements.txt`, `Pipfile`, `setup.py`, etc.).
-- **CheckTests**: Checks for the presence of test directories or test files.
-- **CheckAlmanack**: Runs the [Software Gardening Almanack](https://github.com/software-gardening/almanack) analysis.
+2. **Advanced Analysis**
+   - [Software Gardening Almanack](https://github.com/software-gardening/almanack) analysis
+   - JOSS (Journal of Open Source Software) criteria evaluation
+   - AI-powered repository analysis (optional, requires Synapse agent ID)
+   - Test execution and coverage
 
-The final output is a **consolidated CSV report** where each row represents a tool (i.e., a repository) with the following columns:
+3. **Optional Synapse Integration**
+   - Results upload to Synapse platform
+   - Metadata management
 
-```Tool, CloneRepository, CheckReadme, CheckDependencies, CheckTests, Almanack```
+## Requirements
 
-Each column shows the status (`PASS`/`FAIL`) for the respective check.
+### Core Dependencies
+- Nextflow (version 24.04.3 or later)
+- Docker (required for containerized execution)
+- Python 3.8+
+- Git
 
-## Running the Workflow
-You can execute the workflow in one of two ways:
-- Analyze a single tool by specifying its repository URL.
-- Analyze multiple tools using a sample sheet (CSV file) that includes a repo_url header.
+> [!IMPORTANT]
+> Docker is required to run this workflow. The toolkit uses containerized processes to ensure consistent execution environments across different systems.
 
-### Install Nextflow 
-Follow the official installation guide [here](https://www.nextflow.io/docs/latest/install.html) or use the command below:
+### Optional Dependencies
+For Synapse integration:
+- Synapse Python client
+- Synapse authentication token
+- Synapse configuration file
 
+## Installation
+
+1. **Install Nextflow**
 ```bash
 curl -s https://get.nextflow.io | bash
 ```
 
-### Run with a Single Repository URL
+2. **Install Python Dependencies**
 ```bash
-nextflow run main.nf --repo_url https://github.com/example/repo.git
+pip install -r requirements.txt
 ```
 
-### Run with a Sample Sheet
-Prepare a CSV file (e.g., example-input.csv) with a header repo_url and one URL per row, then run:
+3. **Configure Synapse** (Optional)
+```bash
+# Create Synapse config file
+mkdir -p ~/.synapse
+touch ~/.synapseConfig
+```
 
+> [!NOTE]
+> To use Synapse features, you'll need to:
+> 1. Create a personal access token from your [Synapse Account Settings](https://help.synapse.org/docs/Managing-Your-Account.2055405596.html#ManagingYourAccount-PersonalAccessTokens)
+> 2. Add the token to your `~/.synapseConfig` file:
+>    ```
+>    [authentication]
+>    username = your_username
+>    apiKey = your_personal_access_token
+>    ```
+> 3. Set the token as a Nextflow secret:
+>    ```groovy
+>    // nextflow.config
+>    secrets {
+>        synapse_token = 'your_personal_access_token'
+>    }
+>    ```
```

Comment:
Does this work? I have never seen this way. AFAIK if you are configuring a secret locally you have to use
```
nextflow sercrets set MY_SECRET_KEY MY_SECRET_VALUE
```

You wouldn't really want an access token hanging out in your config anyway